net/HttpRequest.hpp

```diff
@@ -1744,13 +1744,19 @@ class Session final : public ProtocolHandlerInterface
         poll.insertNewSocket(socket);
     }
 
-    void asyncConnect(SocketPoll& poll)
+    void asyncConnect(const std::weak_ptr<SocketPoll>& poll)
     {
         _socket.reset(); // Reset to make sure we are disconnected.
 
-        auto pushConnectCompleteToPoll = [this, &poll](std::shared_ptr<StreamSocket> socket, net::AsyncConnectResult result ) {
-            poll.addCallback([selfLifecycle = shared_from_this(), this, &poll, socket=std::move(socket), result]() {
-                asyncConnectCompleted(poll, socket, result);
+        auto pushConnectCompleteToPoll = [this, poll](std::shared_ptr<StreamSocket> socket, net::AsyncConnectResult result ) {
+            std::shared_ptr<SocketPoll> socketPoll(poll.lock());
```

Comment:
```suggestion
            std::shared_ptr<SocketPoll> socketPoll = poll.lock();
```
Just code style, for consistency. unimportant.